<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PathPlanningWidget</class>
 <widget class="QWidget" name="PathPlanningWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>460</width>
    <height>427</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>350</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>460</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_6">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Path planning</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QListWidget" name="motionList">
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="clearButton">
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="executeButton">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Execute</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="0,0,0,0,0">
     <property name="leftMargin">
      <number>10</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Direction:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="dirBox">
         <item>
          <property name="text">
           <string>forward</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>reverse</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>left</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>right</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Speed: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="speedSpinBox">
         <property name="suffix">
          <string/>
         </property>
         <property name="decimals">
          <number>2</number>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>0.150000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Time: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="timeSpinBox">
         <property name="suffix">
          <string> s</string>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Wait:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="waitSpinBox">
         <property name="suffix">
          <string> s</string>
         </property>
         <property name="decimals">
          <number>1</number>
         </property>
         <property name="singleStep">
          <double>0.500000000000000</double>
         </property>
         <property name="value">
          <double>2.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="addButton">
       <property name="text">
        <string>Add</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLineEdit" name="lineEdit">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="loadPathButton">
       <property name="text">
        <string>Load Path</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>PathPlanningWidget</receiver>
   <slot>onAddSegment()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>163</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>executeButton</sender>
   <signal>clicked()</signal>
   <receiver>PathPlanningWidget</receiver>
   <slot>onExecutePath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>202</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loadPathButton</sender>
   <signal>clicked()</signal>
   <receiver>PathPlanningWidget</receiver>
   <slot>onLoadFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>339</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearButton</sender>
   <signal>clicked()</signal>
   <receiver>PathPlanningWidget</receiver>
   <slot>onClear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>392</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onAddSegment()</slot>
  <slot>onExecutePath()</slot>
  <slot>onLoadFile()</slot>
  <slot>onClear()</slot>
 </slots>
</ui>
